src/main/java/com/axis/jenkins/plugins/eiffel/eiffelbroadcaster/EiffelBroadcasterConfig.java

```diff
@@ -95,67 +101,35 @@ of this software and associated documentation files (the "Software"), to deal
 
     private transient final EventValidator eventValidator = new EventValidator();
 
-    /**
-     * Creates an instance with specified parameters.
-     *
-     * @param enableBroadcaster  if this plugin is enabled
-     * @param serverUri          the server uri
-     * @param userName           the user name
-     * @param userPassword       the user password
-     * @param exchangeName       the name of the exchange
-     * @param virtualHost        the name of the virtual host
-     * @param routingKey         the routing key
-     * @param persistentDelivery if using persistent delivery mode
-     * @param appId              the application id
-     */
-    @DataBoundConstructor
-    public EiffelBroadcasterConfig(boolean enableBroadcaster, String serverUri, String userName, Secret userPassword,
-                            String exchangeName, String virtualHost, String routingKey, boolean persistentDelivery,
-                            String appId, String activityCategories, HostnameSource hostnameSource) {
-        this.enableBroadcaster = enableBroadcaster;
-        this.serverUri = serverUri;
-        this.userName = userName;
-        this.userPassword = userPassword;
-        this.exchangeName = exchangeName;
-        this.virtualHost = virtualHost;
-        this.routingKey = routingKey;
-        this.persistentDelivery = persistentDelivery;
-        this.appId = appId;
-        this.activityCategories.addAll(Util.getLinesInString(activityCategories));
-        this.hostnameSource = hostnameSource;
+    public EiffelBroadcasterConfig() {
+        super.load();
+        EiffelEvent.setSourceProvider(new JenkinsSourceProvider());
```

Comment:
Might be a candidate for an initilizer

Reply:
Yep, that should do it. Not sure what the benefit would be though; we'd still have to call super.load() in the constructor, right? So we get rid of one method call in the ctor and move it to a separate method?